src/app/zap-templates/zcl/data-model/chip/application-basic-cluster.xml

```diff
@@ -28,16 +28,23 @@ limitations under the License.
     <attribute side="server" code="0x0001" define="APPLICATION_VENDOR_ID" type="INT16U" min="0x0000" max="0xFFFF" writable="false" optional="false">vendor id</attribute>
     <attribute side="server" code="0x0002" define="APPLICATION_NAME" type="CHAR_STRING" length="32" writable="false" optional="false">application name</attribute>
     <attribute side="server" code="0x0003" define="APPLICATION_PRODUCT_ID" type="INT16U" min="0x0000" max="0xFFFF" writable="false" optional="false">product id</attribute>
-    <attribute side="server" code="0x0005" define="APPLICATION_ID" type="CHAR_STRING" length="32" writable="false" optional="false">application id</attribute>
-    <attribute side="server" code="0x0006" define="CATALOG_VENDOR_ID" type="INT16U" min="0x0000" max="0xFFFF" writable="false" optional="false">catalog vendor id</attribute>
-    <attribute side="server" code="0x0007" define="APPLICATION_STATUS" type="ENUM8" min="0x00" max="0xFF" default="0x01" writable="false" optional="false">application status</attribute>
+    <attribute side="server" code="0x0004" define="APPLICATION_APP" type="ApplicationBasicApp" writable="true" optional="false">application app</attribute>
+    <attribute side="server" code="0x0005" define="APPLICATION_STATUS" type="ENUM8" min="0x00" max="0xFF" default="0x01" writable="false" optional="false">application status</attribute>
+    <attribute side="server" code="0x0006" define="APPLICATION_VERSION" type="CHAR_STRING" length="32" writable="false" optional="false">application version</attribute>
+    <!-- <attribute side="server" code="0x0007" define="APPLICATION_ALLOWED_VENDOR_LIST" type="ARRAY" entryType="INT16U" length="32" writable="false" optional="false">allowed vendor list</attribute> -->
 
     <command source="client" code="0x00" name="ChangeStatus" optional="false">
       <description>Upon receipt, this SHALL change the running status of the application.</description>
       <arg name="status" type="ApplicationBasicStatus"/>
     </command>
   </cluster>
 
+  <struct name="ApplicationBasicApp">
```

Comment:
Why is this not just called "Application" as in the spec?

Reply:
I'm not sure if it's still case. But before I had to prefix everything with cluster name to make the struct name unique.
I've had conficts before when I've named structs per spec as there were already the structs of the same name somewere else defined.

Reply:
That's why it's named ApplicationBasicApp, and not just application.